src/twisted/mail/_except.py

```diff
@@ -94,25 +96,32 @@ class SMTPClientError(SMTPError):
     """
 
     def __init__(
-        self, code, resp, log=None, addresses=None, isFatal=False, retry=False
+        self,
+        code: int,
+        resp: bytes,
+        log: Optional[bytes] = None,
+        addresses: Optional[object] = None,
+        isFatal: bool = False,
+        retry: bool = False,
     ):
         """
         @param code: The SMTP response code associated with this error.
-
         @param resp: The string response associated with this error.
-
         @param log: A string log of the exchange leading up to and including
             the error.
-        @type log: L{bytes}
-
         @param isFatal: A boolean indicating whether this connection can
             proceed or not. If True, the connection will be dropped.
-
         @param retry: A boolean indicating whether the delivery should be
             retried. If True and the factory indicates further retries are
             desirable, they will be attempted, otherwise the delivery will be
             failed.
         """
+        if isinstance(resp, str):  # type: ignore[unreachable]
+            resp = resp.encode("utf-8")  # type: ignore[unreachable]
+
+        if isinstance(log, str):  # type: ignore[unreachable]
+            log = log.encode("utf-8")  # type: ignore[unreachable]
+
```

Comment:
The old code in `__bytes__` handled `str` input, so I'm doing that normalization here in `__init__`.
The input type remains `bytes` only, since that appears to be the intention.